### PR TITLE
datastore_mongo: set updated_ts on UpsertAttributes

### PR DESCRIPTION
**Pull request by mchalski**
also, selectively set created_ts.

Issues: MEN-649

Signed-off-by: mchalczynski <marcin.chalczynski@rndity.com>

@maciejmrowiec @kjaskiewiczz @bboozzoo @marekswiecznik

### CODE REVIEW

File: datastore_mongo_test.go [hunk 1]
```diff
@@ -634,6 +645,14 @@ func TestMongoUpsertAttributes(t *testing.T) {
 			t.Errorf("attributes mismatch, have: %v\nwant: %v", dev.Attributes, tc.outAttrs)
 		}
 
+		//check timestamp validity
+		//note that mongo stores time with lower precision- custom comparison
+		assert.Equal(t, createdTs.Unix(), dev.CreatedTs.Unix())
```
Comment (bboozzoo) [inline datastore_mongo_test.go]: I used `assert.WithinDuration()` with 1s delta to workaround precision issues. You should be able to replace both asserts with that. Make sure to leave the comment about precision loss.

File: datastore_mongo.go [hunk 2]
```diff
@@ -156,15 +157,20 @@ func (db *DataStoreMongo) UpsertAttributes(id DeviceID, attrs DeviceAttributes)
 	c := s.DB(DbName).C(DbDevicesColl)
 
 	update := makeAttrUpsert(attrs)
-	update = bson.M{"$set": update}
+
+	//set update time and optionally created time
+	now := time.Now()
+	update["updated_ts"] = now
+	update = bson.M{"$set": update,
+		"$setOnInsert": bson.M{"created_ts": now}}
```
Comment (bboozzoo) [inline datastore_mongo.go]: mongo magic ;)

Comment (mchalski) [inline datastore_mongo.go]: indeed, what can you do :)

File: datastore_mongo_test.go (hunk 1 above)
Comment (mchalski) [inline datastore_mongo_test.go]: ...meh, but it won't work in Condition, will still have to keep unix times, or am I missing something

Comment (bboozzoo) [inline datastore_mongo_test.go]: fair enough

**Pull request merged**